src/extension/network/NUClearNetwork.cpp

```diff
@@ -46,10 +46,10 @@ namespace extension {
          *
          * @return the data and who it was sent from
          */
-        std::pair<util::network::sock_t, std::vector<char>> read_socket(fd_t fd) {
+        std::pair<util::network::sock_t, std::vector<uint8_t>> read_socket(fd_t fd) {
 
             // Allocate a vector that can hold a datagram
-            std::vector<char> payload(1500);
+            std::vector<uint8_t> payload(1500);
```

Comment:
For consistency with other places that you construct a vector
```suggestion
            std::vector<uint8_t> payload(1500, 0);
```

Reply:
I specifically don't do that as even though it's minor I shouldn't need to zero it out before use